### DISCUSSION TITLE
Auto Register and Serve Assets

### DISCUSSION THREAD

**Discussion by jpyen** [maddalax/htmgo#53] [General]
Is it possible to auto register routes and serve assets like js, css or others?
Instead of:
```go
h.Script("/public/htmgo.js"),
h.Style(`
    html {
             scroll-behavior: smooth;
            }
   `),
``` 		
Something along the lines of:
```go
  h.Script(assets.UserJS),
  h.Style(assets.LandingPageStyle),
``` 

Similar to how partials are auto-exposed and can be referred to by func name:

```go
h.GetPartial(
   partials.CurrentTimePartial,
   "load, every 1s"),
))
```

**Comment by maddalax** [comment]
Potentially yes, I'll create an issue from this to keep track of it, don't think it'd be too hard. Could generate a const for any file name in the /assets directory (excluding dist)
